plugins/status/plugin.go

```diff
@@ -83,15 +102,15 @@ func (c *Config) validateAndInjectDefaults(services []string) error {
 	}
 
 	// If a service wasn't found, and console logging isn't enabled.
-	if c.Service == "" && !c.ConsoleLogs {
+	if c.Plugin == nil && c.Service == "" && !c.ConsoleLogs {
 		return fmt.Errorf("invalid status config, must have a `service` target or `console` logging specified")
 	}
 
 	return nil
 }
 
 // ParseConfig validates the config and injects default values.
-func ParseConfig(config []byte, services []string) (*Config, error) {
+func ParseConfig(config []byte, services []string, plugins []string) (*Config, error) {
```

Comment:
Instead of changing the public API, we could add a new extensible way to build the status configuration similar to way we build the discovery configuration. See [this](https://github.com/open-policy-agent/opa/blob/master/plugins/discovery/config.go#L35) for reference.

Reply:
We could, I just followed the same change that was made to the decision log plugin.  If I change this one, should I change decision logs too?  Having two different APIs for basically the same functionality seems confusing.

Reply:
I'm thinking if we will break some OPA library integration if we make this change. OTOH looking at the decision logger changes we went the same route as the current change. It would be nice if the decision log had something similar but I think it's fine if that's not part of this change.